GCC 8 on Discover SLES12 cannot build UMD_Etc apps

When trying to build with GCC8 on discover SLES12 I see:
```
[  0%] Built target UMD_utils
[  0%] Linking Fortran executable ../../../../bin/anaice2rst.x
/usr/bin/ld: /discover/swdev/mathomp4/Baselibs/ESMA-Baselibs-5.1.7-SLES12/x86_64-unknown-linux-gnu/gfortran_8.2.0-openmpi_4.0.1-mattbuild-noverbs/Linux/lib/libcurl.a(libcurl_la-curl_threads.o): undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
src/Applications/@UMD_Etc/UMD_utils/CMakeFiles/anaice2rst.x.dir/build.make:88: recipe for target 'bin/anaice2rst.x' failed
make[3]: *** [bin/anaice2rst.x] Error 1
```

--- a/UMD_oletkf/CMakeLists.txt
+++ b/UMD_oletkf/CMakeLists.txt
@@ -28,17 +28,17 @@ target_compile_definitions (${this} PUBLIC _LAPACK_ gmao_intf HAVE_ESMF)
 ecbuild_add_executable (
    TARGET oceanda.x 
    SOURCES letkf.f90 
-   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
+   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} Threads::Threads
    )
 
 ecbuild_add_executable (
    TARGET oceanobsop.x 
    SOURCES obsop.f90 
-   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
+   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} Threads::Threads
    )
 
 ecbuild_add_executable (
    TARGET oceanobs_nc2bin.x 
    SOURCES oceanobs_nc2bin.f90 
-   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
+   LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} Threads::Threads
    )

--- a/UMD_utils/CMakeLists.txt
+++ b/UMD_utils/CMakeLists.txt
@@ -22,7 +22,7 @@ foreach (file ocean_recenter anaice2rst ocean_moments ocean_iau)
    ecbuild_add_executable (
       TARGET ${file}.x 
       SOURCES ${file}.f90 
-      LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
+      LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} Threads::Threads
       )
 endforeach ()
 

--- a/UMD_utils/sosie-2.6.4/CMakeLists.txt
+++ b/UMD_utils/sosie-2.6.4/CMakeLists.txt
@@ -30,7 +30,7 @@ foreach (file sosie corr_vect interp_to_line mask_drown_field)
    ecbuild_add_executable (
       TARGET ${file}.x 
       SOURCES src/${file}.f90 
-      LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES}
+      LIBS ${this} ${NETCDF_LIBRARIES} ${MPI_Fortran_LIBRARIES} Threads::Threads
       )
 endforeach ()
 

